src/SprykerEco/Zed/ComputopApi/Business/Adapter/AbstractAdapter.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\ComputopApi\Business\Adapter;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\RequestOptions;
+use SprykerEco\Zed\ComputopApi\Business\Exception\ComputopApiHttpRequestException;
+use SprykerEco\Zed\ComputopApi\ComputopApiConfig;
+
+abstract class AbstractAdapter implements AdapterInterface
```

Comment:
The 3rd party dependency introduces too low this way. We should try to move it to the dependency\adapter section when it's possible. In the current situation it seems possible. Here is a draft version of the solution:

```
<?php

namespace SprykerEco\Zed\ComputopApi\Dependency\External;

class ComputopApiRequestGuzzleClientAdapter implements ComputopApiRequestClientInterface
{
    protected const DEFAULT_TIMEOUT = 45;

    /**
     * @var ...
     */
    protected $guzzleClient;
    
    public function __construct()
    {
        $this->guzzleClient = new Client([RequestOptions::TIMEOUT => static::DEFAULT_TIMEOUT]);
    }

    /**
     * @param array $options
     *
     * @throws \SprykerEco\Zed\ComputopApi\Business\Exception\ComputopApiHttpRequestException
     *
     * @return []
     */
    protected function send(string $url, array $options = []): array
    {
        try {
            $response = $this->guzzleClient->post($url, $options);
        } catch (RequestException $requestException) {
            throw new ComputopApiHttpRequestException(
                $requestException->getMessage(),
                $requestException->getCode(),
                $requestException
            );
        }

        parse_str($response->getBody()->getContents(), $responseArray);
        
        return $responseArray;
    }

    // ALTERNATIVE
    
    /**
     * @param array $options
     *
     * @throws \SprykerEco\Zed\ComputopApi\Business\Exception\ComputopApiHttpRequestException
     *
     * @return \SprykerEco\Zed\ComputopApi\Dependency\External\ComputopApiClientResponseInterface
     */
    protected function send(string $url, array $options = []): ComputopApiClientResponseInterface
    {
        try {
            $response = $this->guzzleClient->post($url, $options);
        } catch (RequestException $requestException) {
            throw new ComputopApiHttpRequestException(
                $requestException->getMessage(),
                $requestException->getCode(),
                $requestException
            );
        }
        
        return new ComputopApiGuzzleClientResponse($response->getBody());
    }
}
```

Dependency provider:
```
        $container[static::COMPUTOP_API_REQUEST_CLIENT] = function () {
            return $this->createComputopApiRequestClient();
        };

      
        public function createComputopApiRequestClient(): ...Interface
        {
             return new ComputopApiRequestGuzzleClient();
        }
```

Reply:
After this, probably you need to revaluate the "Adapter" namespace's name.